### PR TITLE
Make references work with custom primary keys

### LINKED ISSUE #228: Error with non-ObjectId primary key in referenced field
Hi!

I have the following models:

```python
class Parent(Document):
    child = ReferenceField('Child')


class Child(Document):
    text = StringField(primary_key=True)
```

An `OPTIONS` request on the `Parent` endpoint in the browsable API displays something like:

```
// ...
"child": {
    "type": "field",
    "required": true,
    "read_only": false,
    "label": "Type",
    "choices": [
        {
            "value": "word",
            "display_name": "USLType object"
        }
    ]
}
// ...
```

But when I send this `POST` request:

```
{
    "child": "word"
}
```

I obtain the error:

```
{
    "child": [
        "Cannot parse \"text1\" as ObjectIdField."
    ]
}
```

If I send the request with the id of the `Child`:

```
{
    "child": "5947f817561e3ad0aacfc6be"
}
```

I obtain the Django error:

```
LookUpError at /library/usls/

Cannot resolve field "id"
```

Thanks

**Edit :** explanations [here](https://github.com/umutbozkurt/django-rest-framework-mongoengine/blob/2067a702837a2387c7c1c01b2d296b44a53d25b5/rest_framework_mongoengine/fields.py#L211).

### PR DESCRIPTION
**Pull request by Vayel**
Draft implementation to fix #228. @BurkovBA does the principle of this solution appear correct to you?

### CODE REVIEW

File: rest_framework_mongoengine/fields.py [hunk 1]
```diff
@@ -279,7 +279,7 @@ def to_internal_value(self, value):
             doc_id = self.parse_id(value)
 
         try:
-            return self.get_queryset().only('id').get(id=doc_id).to_dbref()
+            return self.get_queryset().only('pk').get(pk=doc_id).to_dbref()
```
Comment (Vayel) [inline rest_framework_mongoengine/fields.py]: When a model has a custom primary key, the `_fields` attribute of its class does not contain `"id"`.

**Comment by BurkovBA** [comment]
@Vayel 
Vincent, I'm sorry for blinking these last days. I'm on the run with my day-job and side-project.

From what I see so far, this is a great PR! When you feel ready, let me know, I'll review it and merge.

Cheers!

**Comment by Vayel** [comment]
@BurkovBA no problem, thanks for your support!

I think the PR is ready. I made two commits so that you can easily revert the last one and see the new tests failing when the fix is not applied.

**Comment by Vayel** [comment]
@BurkovBA thanks for the merge! How do you decide when you update the PyPI package?

**Comment by BurkovBA** [comment]
@Vayel Hi, Vincent, awesome contributions, thanks.

Well, DRF-ME is a classical example of [sentimental versioning](http://sentimentalversioning.org/) - the version of package was meant to be in sync with DRF. As we're breaking with DRF 3.5 or 3.6, I keep releasing under 3.3.* major/minor series.

I think, we created enough new features to do a release, so I need to list them and at least mention in documentation and then will roll out the next release. Hopefully, I can do it this week.

Actually, could you write to Umut Bozkurt and ask him to add you to the list of maintainers of this project? Cause clearly you have more time to contribute to it then I do of late and you have a deep understanding of its internals.

**Pull request merged**